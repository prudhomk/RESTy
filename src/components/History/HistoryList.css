.History{
    width: 17.5vw;
    margin: 5px;
    display: flex;
    flex-direction: column;
    margin-right: 5px;
}

.History li {
      background-color: orange;
      color: white;
      width: 17vw;
      font-size: 13.5px;
      margin: 5px;
      padding: 5px;
      border-radius: 5px;
      border: 1px solid lightgray;
      text-overflow: clip;
}

  
  