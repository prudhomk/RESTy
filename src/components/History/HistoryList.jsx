/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import History from './History';

const HistoryList = ({ history, onClick }) => {
  const historyDetails = history.map((item, i) => {
    return ( 
      <li key={i} onClick={onClick}>
        <History method={item.method} url={item.url} id={`${item.url}+${item.method}`} />
      </li>
    );
  });

  return (
    <ul>
      {historyDetails}
    </ul>
  );
};

HistoryList.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired,
  })).isRequired,
  onClick: PropTypes.func.isRequired
};

export default HistoryList;
