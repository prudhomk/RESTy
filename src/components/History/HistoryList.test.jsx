/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from '../History/HistoryList';

describe('HistoryList component', () => {
  const history = [{ 'url':'', 'method':'PUT', 'body':'', 'key':'+PUT' }, { 'url':'', 'method':'PATCH', 'body':'booger', 'key':'+PATCH' }, { 'url':'https://ac-vill.herokuapp.com/villagers', 'method':'GET', 'body':'', 'key':'https://ac-vill.herokuapp.com/villagers+GET' }]
  const onClick = jest.fn();
  afterEach(() => cleanup());
  it('renders HistoryList', () => {
    const { asFragment } = render(<HistoryList history={history} onClick={onClick}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
