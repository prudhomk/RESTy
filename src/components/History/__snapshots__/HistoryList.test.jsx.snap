// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component renders HistoryList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <article
        id="+PUT"
      >
        <p
          id="+PUT"
        >
          PUT
        </p>
        <span
          id="+PUT"
        />
      </article>
    </li>
    <li>
      <article
        id="+PATCH"
      >
        <p
          id="+PATCH"
        >
          PATCH
        </p>
        <span
          id="+PATCH"
        />
      </article>
    </li>
    <li>
      <article
        id="https://ac-vill.herokuapp.com/villagers+GET"
      >
        <p
          id="https://ac-vill.herokuapp.com/villagers+GET"
        >
          GET
        </p>
        <span
          id="https://ac-vill.herokuapp.com/villagers+GET"
        >
          https://ac-vill.herokuapp.com/villagers
        </span>
      </article>
    </li>
  </ul>
</DocumentFragment>
`;
